commands/init.go

```diff
@@ -315,23 +316,34 @@ func (a *initFnCmd) init(c *cli.Context) error {
 func runInitImage(initImage string, a *initFnCmd) error {
 	fmt.Println("Building from init-image: " + initImage)
 
-	// Run the initImage
-	var c1ErrB bytes.Buffer
-	tarR, tarW := io.Pipe()
+	var stdoutBuf, stderrBuf bytes.Buffer
+	var errStdout, errStderr error
 
-	c1 := exec.Command("docker", "run", "-e", "FN_FUNCTION_NAME="+a.ff.Name, initImage)
-	c1.Stderr = &c1ErrB
-	c1.Stdout = tarW
+	cmd := exec.Command("docker", "run", "-e", "FN_FUNCTION_NAME="+a.ff.Name, initImage)
+	stdoutIn, _ := cmd.StdoutPipe()
+	stderrIn, _ := cmd.StderrPipe()
 
-	c1Err := c1.Start()
-	if c1Err != nil {
-		fmt.Println(c1ErrB.String())
-		return errors.New("Error running init-image")
+	stdout := io.Writer(&stdoutBuf)
+	stderr := io.Writer(&stderrBuf)
+
+	err := cmd.Start() // can start ever fail?
+
+	go func() {
+		_, errStdout = io.Copy(stdout, stdoutIn)
+	}()
+
+	go func() {
+		_, errStderr = io.Copy(stderr, stderrIn)
+	}()
+
+	err = cmd.Wait()
+	if err != nil || errStdout != nil || errStderr != nil {
+		return errors.New("Error, unbale to use the init-image! Make sure that Docker is running and that the init-image exists.")
```

Comment:
think I'm grokking what's going on, but we shouldn't need the two goroutines, and we can call wait at the end regardless of whether untarStream returns an error or not? ie we need to check that the command succeeded regardless of whether the untar succeeds, and from there we can make the useful error. if the command succeeds and untar succeeds, we're good. I think the pipe was preferable to a buffer, this adds a good bit of memory overhead as images can be pretty large (few megs, at least). basically asking to move wait until after untarStream and check cmd.Wait instead of bailing if untarStream has an error (and if untarStream does have an error but cmd.Wait does not, return untarStream's error?)? not sure if that works though?